src/api/rate_limiter.py

```diff
@@ -78,31 +80,81 @@ async def handle_requests(self):
             if request.method == "GET":
                 # keep track of the last time a request was made
                 last_time_request = datetime.now()
+
                 # actually send the GET request
-                resp = requests.get(request.url, cookies=request.cookies)
-                if resp.status_code != 200:
-                    self.logger.warning(
-                        "Request : GET %s + %s failed with response code %s",
+                try:
+                    resp = requests.get(request.url, cookies=request.cookies, timeout=DEFAULT_MAX_TIMEOUT)
+                except requests.exceptions.Timeout:
+                    self.logger.error("Request GET %s + %s: the request timed out", 
                         request.url,
                         request.cookies,
-                        resp.status_code,
+                        retry_count, 
+                        self._max_retry
                     )
                     if retry_count < self._max_retry:
                         retry = True
                         retry_count += 1
-                    else:
+                        continue
+                    
+                    self.logger.error("Request timed out too many times. Potential ban")
+                    
+                    # TODO Send message on discord
+                    
+                    self.logger.error("Waiting 10min.")
+                    await asyncio.sleep(600)
+                    self.logger.warning("Waking up: ready to retry after timeout")
+                    retry = True
+                    retry_count += 1
+                    continue
+
+                if resp.status_code == 429:
+                    try:
+                        wait = int(resp.headers['Retry-After'])
+                    except:
                         self.logger.error(
-                            "Failed to get request after %s attempt. We could be banned :(",
-                            self._max_retry,
+                            "Request got 429 and failed to parse headers: %s",
+                            resp.headers
                         )
-                        raise RuntimeError("Looks like a ban to me :'(")
-                else:
-                    data = resp.json()
+                        raise RuntimeError("Wrong headers")
```

Comment:
pylint will probably issue a warning saying something like "too global exception"
you should use :
```py
except Exception as exc:
    ...
    raise RuntimeError("blabla") from exc
```
or even better a custom exception?

or maybe let me deal with it when adding the message sending in discord, I'll probably have to mess with the exception raised